src/main/java/competition/subsystems/turret/commands/TurretRotateToVisionTargetCommand.java

```diff
@@ -0,0 +1,64 @@
+package competition.subsystems.turret.commands;
+
+import com.google.inject.Inject;
+import competition.subsystems.turret.TurretSubsystem;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import xbot.common.command.BaseCommand;
+
+public class TurretRotateToVisionTargetCommand extends BaseCommand
+{
+    final TurretSubsystem turretSubsystem;
+    final NetworkTable ambanNetworkTable;
+
+    @Inject
+    public TurretRotateToVisionTargetCommand(TurretSubsystem tSubsystem, NetworkTableInstance networkTableInstance) {
+        this.turretSubsystem = tSubsystem;
+        this.addRequirements(this.turretSubsystem);
+        this.ambanNetworkTable = networkTableInstance.getTable("amban");
+    }
+
+    @Override
+    public void execute() {
+        // Check if vision subsystem is active and our turret is calibrated
+        if (isAmbanActive() && isTargetLocked())
+        {
+            // yawToTarget may ask us to go beyond our rotation limit so we need to correct for this
```

Comment:
This would be a good place to add a `LoggingLatch` - so we can write a log line each time we go from "not seeing the target" to "target lock"

Reply:
I think i've done this correctly